pkg/controllers/dashboardapi/nodedriver/nodedriver.go

```diff
@@ -0,0 +1,61 @@
+package nodedriver
+
+import (
+	"context"
+	"sync/atomic"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/drivers"
+	"github.com/rancher/rancher/pkg/wrangler"
+)
+
+var leader = atomic.Bool{}
+
+func Register(ctx context.Context, wrangler *wrangler.Context) {
+	wrangler.Mgmt.NodeDriver().OnChange(ctx, "custom-node-driver-handler", onChange)
+
+	// when this pod becomes the leader, do not run the onChange code
+	wrangler.OnLeader(func(ctx context.Context) error {
+		leader.Store(true)
+		return nil
+	})
+}
+
+func onChange(_ string, obj *v3.NodeDriver) (*v3.NodeDriver, error) {
```

Comment:
Maybe it would be pointless (I'm curious what your take is), but could it be worth writing a small unit test for this?

Reply:
I can't really write a unit test that confirms expected behavior, but I can probably  write one that verifies we don't process invalid drivers.